feat: added a paste option for environment variables on app deployment

katushabeannet opened the pull request:
# Description

This PR adds functionality to paste multiple environment variables at once from .env file format in the application deployment form. The implementation includes:

A new "Paste Variables" button that opens a modal dialog

Modal with textarea for pasting .env content and clear formatting instructions

Logic to parse and validate the pasted variables

## Type of change
- [ ] New feature (non-breaking change which adds functionality)


## Trello Ticket ID

Please add a link to the Trello ticket for the task.

## How Can This Been Tested?

1.Navigate to the application deployment form

2.Click "Paste Variables" button in the Environment Variables section

3.In the modal that appears:

Paste content in .env format (KEY=VALUE per line)

Try various formats (valid, invalid, empty lines)

Test the Cancel button

Test the Add Variables button

4.Verify:

Variables are parsed correctly

First empty position is filled if available

Invalid lines are skipped

State is properly maintained

UI remains responsive


## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

# Screenshots
<img width="642" height="161" alt="Capture" src="https://github.com/user-attachments/assets/7d849ea7-3afc-4811-8f35-0cb8868f91f6" />
<img width="619" height="313" alt="paste" src="https://github.com/user-attachments/assets/8bdebb07-ce9c-4f81-83a7-36ebdf17c66c" />

katushabeannet committed:
feat: added a paste option for environment variables on app deployment

rhodinemma commented on src/components/Forms/CreateAppForm.tsx:
```diff
@@ -110,6 +114,41 @@ export const CreateSingleAppForm = (props: {
+      .map((line) => {
+        const [key, ...values] = line.split("=");
+        // eslint-disable-next-line curly
+        if (!key || values.length === 0) return null;
```
you should always let the code linter show you how to fix such, or you also run `yarn run prettier:write` in your terminal, instead of adding these eslint disable comments.

rhodinemma commented on src/components/Forms/CreateAppForm.tsx:
```diff
@@ -110,6 +114,41 @@ export const CreateSingleAppForm = (props: {
+
+  const addVariablesFromPaste = () => {
+    // eslint-disable-next-line curly
+    if (!pastedContent) return;
```
even this can be addressed by the available code linter.

rhodinemma reviewed (changes requested):
good work on this @katushabeannet, kindly use the code formatter as shared above to format all your code changes.

katushabeannet commented:
ok point taken

On Fri, Jul 11, 2025 at 1:45 PM Rhodin Nagwere ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> good work on this @katushabeannet <https://github.com/katushabeannet>,
> kindly use the code formatter as shared above to format all your code
> changes.
> ------------------------------
>
> In src/components/Forms/CreateAppForm.tsx
> <https://github.com/crane-cloud/frontend/pull/999#discussion_r2200336119>:
>
> > @@ -110,6 +114,41 @@ export const CreateSingleAppForm = (props: {
>      setEnvVariables([...envVariables, { key: "", value: "" }]);
>    };
>
> +  const parseVariablesFromContent = (content: string) => {
> +    const lines = content.split("\n");
> +    return lines
> +      .map((line) => {
> +        const [key, ...values] = line.split("=");
> +        // eslint-disable-next-line curly
> +        if (!key || values.length === 0) return null;
>
> you should always let the code linter show you how to fix such, or you
> also run yarn run prettier:write in your terminal, instead of adding
> these eslint disable comments.
> ------------------------------
>
> In src/components/Forms/CreateAppForm.tsx
> <https://github.com/crane-cloud/frontend/pull/999#discussion_r2200337939>:
>
> > +    return lines
> +      .map((line) => {
> +        const [key, ...values] = line.split("=");
> +        // eslint-disable-next-line curly
> +        if (!key || values.length === 0) return null;
> +        return {
> +          key: key.trim(),
> +          value: values.join("=").trim(),
> +        };
> +      })
> +      .filter(Boolean) as { key: string; value: string }[];
> +  };
> +
> +  const addVariablesFromPaste = () => {
> +    // eslint-disable-next-line curly
> +    if (!pastedContent) return;
>
> even this can be addressed by the available code linter.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/crane-cloud/frontend/pull/999#pullrequestreview-3009672798>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJJC734FCF5C4FHFVWZFZFD3H6IVZAVCNFSM6AAAAACBJKBJJGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTAMBZGY3TENZZHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

katushabeannet committed:
feat: added fuctionality to handle invalid variable input

Mubangizi reviewed (approved):
Great work on this @katushabeannet
